consensus/hotstuff/notifications/telemetry.go

```diff
@@ -217,7 +246,7 @@ func (p *PathHandler) CloseCurrentPath() *PathHandler {
 	return p
 }
 
-// IsCurrentPathOpen if and only is the most recently started path has been closed.
+// IsCurrentPathClosed if and only is the most recently started path has been closed.
 func (p *PathHandler) IsCurrentPathClosed() bool {
```

Comment:
I would like to take another approach. When doing this I've tried to design it in a way that there is always path. We should be able to satisfy this condition since hotstuff related state(forks, pacemaker) should be modified ONLY from `EventHandler`. Meaning we always know where the path starts and ends. The way it is right now it should be the case. I would be extremely interested to see such error and to find out where we break this assumption.

Reply:
Here's an example from Mainnet: https://dapperlabs.grafana.net/explore?orgId=1&left=%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22editorMode%22:%22code%22,%22expr%22:%22%7Bnetwork%3D%5C%22mainnet20%5C%22,host%3D%5C%22consensus-001%5C%22%7D%20%7C%3D%20%5C%22no%20path%5C%22%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-2d%22,%22to%22:%22now%22%7D%7D. We get the no path errors pretty consistently when the node starts up.

Reply:
Right, I've specifically addressed this by capturing all events into path. For startup you can check this commit: https://github.com/onflow/flow-go/pull/3571/commits/d1da95413d1a06b901dd1f9efc749657e597b9b5

Reply:
OK nice!

Reply:
Thanks Yurii. Great work fixing these error logs!